llrt_core/src/vm.rs

```diff
@@ -423,10 +423,16 @@ impl Vm {
         let ctx = AsyncContext::full(&runtime).await?;
         ctx.with(|ctx| {
             for init_global in init_globals {
-                init_global(&ctx)?;
+                if let Err(err) = init_global(&ctx).catch(&ctx) {
```

Comment:
Since we’re handling the error the same way we can put everything in a closure and handle the error once:

```rust
({
//all init code
}).catch()…
```

Reply:
Thanks for pointing that out. It was a little confusing, but there was a similar implementation, so we modified it based on the following. :)

https://github.com/awslabs/llrt/blob/9824e4163fb8cf103066d7b5938929740304a4ff/llrt_core/src/compiler.rs#L65-L66